pkg/components/disk_manifest_loader.go

```diff
@@ -118,7 +131,7 @@ func (m DiskManifestLoader[T]) decodeYaml(b []byte) ([]T, []error) {
 			continue
 		}
 
-		manifest := m.zv()
+		var manifest T
```

Comment:
you must be thinking: why the hell did someone complicated a zero value instantiation in that way?

Not saying that you did the wrong thing, but let me tell the why: when I created this `DiskManifestLiader` I tried to kept the intention of the user that created the `newComponent` method, allowing folks to set default fields on the component "zero" value. Obviously, that given the current state of the `newComponent` method it can be completely replaced by what you did, but keep in mind that you have removed the possibility of setting default fields for whatever is unmarshalled

> There's workarounds to do the same like implementing a custom unmarshaller, however I'd prefer the old way :)

Reply:
If we want to keep the old way of doing it then we should add a really wordy comment around it so people don't change it w/o really thinking hard about the implications. You won't always be around to add this commentary to every PR forever :-)

Reply:
@mcandeia Thanks for explaining. I have re-added the ability to set a function that returns a zero value. In a slightly different way, to not alter the signature of `NewDiskManifestLoader` :)